client/src/components/User.jsx

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import OnHover from "./OnHover.jsx";
+
+class User extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
```

Comment:
This is a really large component. Have you considered breaking it up into more smaller components? Will make it easier to read and debug/test.